lib/Conversion/TritonGPUToLLVM/Utility.h

```diff
@@ -13,13 +13,17 @@
 #define inttoptr(...) rewriter.create<LLVM::IntToPtrOp>(loc, __VA_ARGS__)
 #define ptrtoint(...) rewriter.create<LLVM::PtrToIntOp>(loc, __VA_ARGS__)
 #define zext(...) rewriter.create<LLVM::ZExtOp>(loc, __VA_ARGS__)
+#define trunc(...) rewriter.create<LLVM::TruncOp>(loc, __VA_ARGS__)
```

Comment:
@rsanthanam-amd changes made in this file doesn't seem related with bfloat16 support to me. could you help elaborate a bit?

Reply:
i added these shorthand macros to this file because they were useful for the AMDGPU bfloat16 implementation and also because others may find them useful in future contexts.

for example, the zext macro was already there and it was useful for this PR.

In a similar fashion, I added trunc() and the shift macros.

from ElementwiseOpToLLVM.cpp:

auto shifted = shl(i32_ty, as_int32, i32_val(16));

// ...

auto shifted = lshr(i32_ty, as_int32, i32_val(16));
auto truncated = trunc(i16_ty, shifted);